bool first
---
string hint
string which_hint
bool last

